modules/exploits/multi/veritas/beagent_sha_auth_rce.rb

```diff
@@ -0,0 +1,536 @@
+# frozen_string_literal: true
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Tcp
+  include Msf::Exploit::Remote::NDMPSocket
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::EXE
```

Comment:
```suggestion
  include Msf::Exploit::EXE
  prepend Msf::Exploit::Remote::AutoCheck
```

You've written a great check method - I'd suggest prepending AutoCheck here so the method runs automatically before attempting to exploit.